### DISCUSSION TITLE
Test data on TimeSeries

### DISCUSSION THREAD

**Discussion by michelidiego** [autogluon/autogluon#3378] [Q&A] [answered]
Hi,

I was trying to get familiar with the Timeseries forecast and I have a question.

When I say the video introducing this models it says that you need to use the entire time series as data to train the model, but when I'm looking at the example provided in the Quick Start I see that you use train data and test data.

my question is: The test data is only required for plotting purposes? or is there a different usage like cross validation or something like that?

thank you for your help

**Comment by shchur** [comment, marked as answer]
Hi, you are correct, AutoGluon automatically generates the validation set based on the training data that you provided. 

In the [Quick Start Tutorial](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-quick-start.html) we only use the test data to evaluate the trained `TimeSeriesPredictor` on some unseen data. You can split your data into train/test sets using [these instructions](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#how-to-evaluate-forecast-accuracy), but this is purely optional and is not required for fitting the predictor.

**Reply by michelidiego** [reply]
Thank you for the comprehensive answer.

**Reply by ArijitSinghEDA** [reply]
I have a training set in the format:

col1 | col2 | col3 | target
--- | --- | --- | ---
val1 | val2 | val3 | targ1
val1 | val2 | val3 | targ2

I have passed this in a `TimeSeriesDataFrame` and trained my model using `TimeSeriesPredictor`.

Now, I have a test dataset in the format:

col1 | col2 | col3
--- | --- | ---
val1 | val2 | val3
val1 | val2 | val3

When I try to predict the result on this data set using `TimeSeriesPredictor.predict()`, it throws an error:
`KeyError: "Only a column name can be used for the key in a dtype mappings argument. 'target' not found in columns."`

Why is this an error? Shouldn't the test dataset be in this format?

**Reply by shchur** [reply]
@ArijitSinghEDA when generating predictions with covariates that are known into the future
```python
predictor.predict(data, known_covariates=known_covariates)
```
you should provide the following data:
- `data` - historic data (up to the start of the forecast horizon), including columns `col1`, `col2`, `col3`, `target`
- `known_covariates` - future values of the known covariates (covering the forecast horizon), including columns `col1`, `col2`, `col3`.

Please have a look at  [this section](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-indepth.html#time-varying-covariates) of the documentation for an example. Specifically, check out the index/columns of `train_data` and `known_covariates`.
